### PR TITLE
feat(create-password): first time access before create account screen.

### PR DESCRIPTION
**Pull request by elenitaex5**
# Description

First time access before create account screen has been added.
This flows has been tested and developed:
- Deep link with no token: redirect to login
- Expired token: shows message with expired activation link
- Token well formed: shows create account screen.
- Token already used: shows screen with already used message and go to login
- Wrong platform to create account: shows a message with wrong platform.

# Checklist:

- [X] Storybook has been updated (if needed) and still working
- [X] Tests have been run locally before create the PR

---

- [ ] Tested in Android simulator
- [ ] Tested in iOS simulator
- [X] Tested in Android device through Expo Go
- [ ] Tested in Android device as app
- [ ] Tested in iOS device through Expo Go
- [ ] Tested in iOS device as app
- [ ] Tested in web

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -19,6 +20,7 @@ Sentry.init({
   dsn: SENTRY_DSN,
   environment: STAGE,
   enableInExpoDevelopment: true,
+  // enableNative: false,
```
Comment (elenitaex5) [inline App.tsx:23]: I know is commented

File: src/contexts/AuthContext.tsx [hunk 2]
```diff
@@ -64,23 +65,33 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
...
+    //     .catch((e) => {
+    //       setIsAuth(false)
+    //       console.log('Error : CurrentAuth ::', e)
+    //     })
+    //     .finally(() => setTimeout(() => SplashScreen.hideAsync(), 1000))
+    //   AsyncStorage.getItem('@abi_locale').then((locale) => {
+    //     if (locale !== null) {
+    //       setLanguage(locale)
+    //     }
+    //   })
```
Comment (ylagef) [inline src/contexts/AuthContext.tsx:93]: Needed?

File: src/helpers/axios-interceptors.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        service: 'API Rest Response',
+        message: JSON.stringify(response.data)
+      })
+      return response
+    },
+    (error) => {
+      Logger(ConsoleLogType.ERROR, {
+        service: 'API Rest',
+        message: `HTTP ${error?.response?.status} - ${JSON.stringify(error?.response?.data)}`,
+        error: error
```
Comment (ylagef) [inline src/helpers/axios-interceptors.ts:28]:
```suggestion
        error
```

File: src/navigation/types.ts [hunk 4]
```diff
@@ -15,8 +16,8 @@ import { NativeStackScreenProps } from '@react-navigation/native-stack'
 export type AuthParamList = {
   LogIn: undefined
   ForgotPassword: undefined
-  CreatePassword: undefined
-  ActionFeedback: undefined
+  CreatePassword: { token: string } | undefined
+  ActionFeedback: { status: ActionFeedbackStatus } | undefined
```
Comment (ylagef) [inline src/navigation/types.ts:20]: Is this the same?

```suggestion
  CreatePassword?: { token: string }
  ActionFeedback?: { status: ActionFeedbackStatus }
```

File: src/screens/CreatePassword/index.tsx [hunk 5]
```diff
@@ -1,16 +1,88 @@
...
+    }
+
+    auth({ email, password, firstAccess: true })
+      .then((session) => {
+        const firstAccess = getAuthInfo(session?.token)
+        if (firstAccess?.newPasswordRequired === false) {
+          navigation.navigate('LogIn')
+          return
+        }
+        return
```
Comment (ylagef) [inline src/screens/CreatePassword/index.tsx:50]: Returns are not needed if they return void, right?

```suggestion
        }
```

File: src/screens/CreatePassword/index.tsx [hunk 6]
```diff
@@ -1,16 +1,88 @@
...
+          setFormErrors({
+            user: { message: Platform.OS === 'web' ? t('new_password.no_permission.webapp') : t('new_password.no_permission.nativeapp') }
+          })
+          return
+        }
+        if (error?.code === 'ERR-AUTH-003') {
+          //User was already activated
+          navigation.navigate('ActionFeedback', { status: ActionFeedbackStatus.PASSWORD })
+          return
+        }
```
Comment (ylagef) [inline src/screens/CreatePassword/index.tsx:70]: Maybe would be nice to have all this error codes moved to an enum in order to reuse it and avoid typos. Not required now, but recommended.

File: src/screens/CreatePassword/index.tsx [hunk 7]
```diff
@@ -1,16 +1,88 @@
...
+      {!expired ? (
+        <Welcome subtitle={t('new_password.create_password')} />
+      ) : (
+        <Welcome testID="expired-welcome" title={t('activation_expired.title')} subtitle={t('activation_expired.description')} />
+      )}
+      {!expired ? (
+        <PasswordCreationForm email={email} formErrors={formErrors} />
+      ) : (
+        <Button text={t('button.login')} onPress={() => linkTo('/login')} testID="expired-button-redirect" />
+      )}
```
Comment (ylagef) [inline src/screens/CreatePassword/index.tsx:85]: Duplicated same condition?

**Review by ylagef — CHANGES_REQUESTED** [review]

File: src/contexts/AuthContext.tsx (hunk 2 above)
Comment (elenitaex5) [reply src/contexts/AuthContext.tsx:93]: YES

File: src/screens/CreatePassword/index.tsx (hunk 5 above)
Comment (elenitaex5) [reply src/screens/CreatePassword/index.tsx:50]: right

File: src/screens/CreatePassword/index.tsx (hunk 6 above)
Comment (elenitaex5) [reply src/screens/CreatePassword/index.tsx:70]: done

**Review by ylagef — APPROVED** [review]

File: src/hooks/useCachedResources.ts [hunk 8]
```diff
@@ -22,7 +22,7 @@ export default function useCachedResources() {
         console.warn(e)
       } finally {
         setLoadingComplete(true)
-        SplashScreen.hideAsync()
+        // SplashScreen.hideAsync()
```
Comment (ylagef) [inline src/hooks/useCachedResources.ts:25]: Can be removed?

File: src/screens/CreatePassword/index.tsx [hunk 9]
```diff
@@ -1,16 +1,89 @@
...
+        if (error?.code === AUTH_ERRORS.PLATFORM_PERMISSION) {
+          setFormErrors({
+            user: { message: Platform.OS === 'web' ? t('new_password.no_permission.webapp') : t('new_password.no_permission.nativeapp') }
+          })
+          return
+        }
+        if (error?.code === AUTH_ERRORS.USER_ALREADY_ACTIVATED) {
+          navigation.navigate('ActionFeedback', { status: ActionFeedbackStatus.PASSWORD })
+          return
+        }
```
Comment (ylagef) [inline src/screens/CreatePassword/index.tsx:67]: Is all ifs check the same variable a switch is better

File: src/screens/CreatePassword/index.tsx [hunk 10]
```diff
@@ -1,16 +1,89 @@
...
+    if (exp < Date.now() / 1000) {
+      Logger(ConsoleLogType.ERROR, { service: 'Create Account', error: new Error(`Create account token has been expired for user ${email}`) })
+      setExpired(true)
+      return
+    }
+
+    auth({ email, password, firstAccess: true })
+      .then((session) => {
+        const firstAccess = getAuthInfo(session?.token)
+        if (firstAccess?.newPasswordRequired === false) {
```
Comment (ylagef) [inline src/screens/CreatePassword/index.tsx:48]: Why don't do this?

```suggestion
        if (!firstAccess?.newPasswordRequired) {
```

**Review by ylagef — COMMENTED** [review]
Check my comments

File: src/hooks/useCachedResources.ts (hunk 8 above)
Comment (elenitaex5) [reply src/hooks/useCachedResources.ts:25]: yes

**Pull request merged**